hazelcast/src/main/java/com/hazelcast/jet/impl/connector/ReadFilesP.java

```diff
@@ -182,6 +185,11 @@ public int preferredLocalParallelism() {
         public FileTraverser<T> traverser() {
             return new LocalFileTraverser<>(LOGGER, directory, glob, ignoreFileNotFound, path -> true, readFileFn);
         }
+
+        @Override
+        public Permission getRequiredPermission() {
+            return ConnectorPermission.file(directory, ACTION_READ);
```

Comment:
This covers only reading via ReadFilesP, if hadoop is used this is not checked. Hadoop (and cloud) have their own permissions, but you can also use hadoop to read local files, effectivelly bypassing this.